CMakeLists.txt

```diff
@@ -54,24 +53,30 @@ add_library(graphqlservice
   GraphQLResponse.cpp
   GraphQLService.cpp
   Introspection.cpp
-  ${CMAKE_BINARY_DIR}/graphqlservice/IntrospectionSchema.cpp)
+  IntrospectionSchema.cpp)
 target_link_libraries(graphqlservice PRIVATE taocpp::pegtl)
 target_link_libraries(graphqlservice PUBLIC Threads::Threads)
-target_include_directories(graphqlservice
-  PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
-    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
-    $<INSTALL_INTERFACE:include>
-)
+# Make system includes (e.g. <graphqlservice/public.h>) work relative to the build/install generator expressions
+target_include_directories(graphqlservice SYSTEM PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+# Make local includes (e.g. "private.h") work while building graphqlservice, but don't export them
+target_include_directories(graphqlservice PRIVATE
```

Comment:
If my previous comment about `SYSTEM` above is right, then you should not need this second `target_include_directories`, has a `PUBLIC` visibility is just the combination of `PRIVATE|INTERFACE`.